File: src/main/java/no/nav/fo/veilarbregistrering/autorisasjon/AutorisasjonService.java
/*
package no.nav.fo.veilarbregistrering.autorisasjon;

import no.nav.common.auth.oidc.OidcTokenValidator;
import no.nav.fo.veilarbregistrering.config.PepConfig;
import org.springframework.stereotype.Component;

@Component
public class AutorisasjonService {



        @Inject
        Provider<HttpServletRequest> httpServletRequestProvider;

        @Inject
        AktorService aktorService;

        @Inject
        PepClient pepClient;

        @Inject
        Pep pep;

        @Inject
        AbacServiceConfig abacServiceConfig;

        private OidcTokenValidator oidcTokenValidator = new OidcTokenValidator();
        private IssoOidcProvider issoProvider = new IssoOidcProvider();



        public void sjekkLesetilgangTilBruker(String fnr) {
            AktorId aktorId = getAktorIdOrElseThrow(aktorService, fnr);
            pepClient.sjekkLesetilgangTilAktorId(aktorId.getAktorId());
        }

    }
}
*/

File: src/main/kotlin/no/nav/fo/veilarbregistrering/autorisasjon/AutorisasjonService.kt
package no.nav.fo.veilarbregistrering.autorisasjon

import no.nav.fo.veilarbregistrering.bruker.Bruker
import no.nav.fo.veilarbregistrering.bruker.Foedselsnummer

interface AutorisasjonService {

    fun sjekkLesetilgangTilBrukerMedNivå3(bruker: Bruker, kontekst: String)
    fun sjekkLesetilgangTilBruker(fnr: Foedselsnummer)
    fun sjekkLesetilgangTilBruker(bruker: Bruker, kontekst: String)
    fun sjekkSkrivetilgangTilBruker(fnr: Foedselsnummer)
    fun sjekkSkrivetilgangTilBruker(bruker: Bruker, kontekst: String)

    fun sjekkSkrivetilgangTilBrukerForSystembruker(fnr: Foedselsnummer, cefMelding: CefMelding)

    fun erVeileder(): Boolean

    val innloggetVeilederIdent: String
}